### PR TITLE
Wwu/alp 74 configure catch 2

### PR DESCRIPTION
**Pull request by WillWu88**
This should be our last batches of manual PR that needs reviewing. Once these tests makes it into the main branch, we would be able to use these tests in our CI pipeline.

### CODE REVIEW

File: src/TestClass/TestClass_test.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include <catch2/catch_test_macros.hpp>
+#include "test.h"
+
+
+TEST_CASE("test the first function") {
```
Comment (Benned-H) [inline src/TestClass/TestClass_test.cpp:5]: We could encourage some style on tests, depending on the complexity. For example, we could state the high-level desired behavior as a full sentence, although for these short-lived examples that's definitely overkill.

**Review by Benned-H — COMMENTED** [review]
Left two comments, they may have been pending but should now show.

File: src/TestClass/TestClass_test.cpp (hunk 1 above)
Comment (WillWu88) [reply src/TestClass/TestClass_test.cpp:5]: Yeah I wrote these at 12am, so not the best tests. This is just a proof of concept for now. What you described can live in a doc string?

File: src/TestClass/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+                           "${PROJECT_SOURCE_DIR}/include"
+                         )
+target_link_libraries(FastSLAM Eigen3::Eigen)
+
+if(BUILD_TESTS)
+  add_executable(test_TestClass TestClass_test.cpp)
+  target_link_libraries(test_TestClass
+                        PRIVATE Catch2::Catch2WithMain
+                        Eigen3::Eigen)
+  catch_discover_tests(test_TestClass)
```
[the remark this answers is not in the document]
Comment (Benned-H) [reply src/TestClass/CMakeLists.txt:14]: Yes, see [Jira](https://willwu.atlassian.net/browse/ALP-74) for comment documentation.

**Review by Benned-H — APPROVED** [review]
LGTM!

**Pull request merged**